extensions/amp-analytics/0.1/transport-plugins.js

```diff
@@ -48,16 +51,25 @@ export const BatchingPluginFunctions = {
 //  * @param {Array<!batchSegmentDef>} batchSegments
 //  * @return {string}
 //  */
-// function ping(baseUrl, batchSegments) {}
+// function ping(baseUrl, batchSegments, isBatchEnabled, useBody) {}
 
 /**
- * @param {string} unusedBaseUrlForTesting
- * @param {Array<!batchSegmentDef>} unusedBatchSegmentsForTesting
+ * The default way for merging batch segments
+ *
+ * @param {string} baseUrl
+ * @param {!Array<!batchSegmentDef>} batchSegments
  * @return {string}
  */
-function ping(unusedBaseUrlForTesting, unusedBatchSegmentsForTesting) {
-  if (getMode().localDev || getMode().test) {
-    return 'testFinalUrl';
+export function defaultPlugin(baseUrl, batchSegments) {
+  const extraUrlParamsStr = batchSegments
```

Comment:
Yep, in next PR. I'm thing of making the return a Class, so we can have all those different methods, and type-checked. What you think?

Reply:
With an @interface? 

One thing I want to be clear is that currently the `batchPlugin` is only used to define the batching serializing algorithm. And later on we can expand it to include the logic to create batched array body object. 
However from what I understand, the class you're imaging also contains logic to handle serialize non-batched request logic. I'm ok with you creating a class now without any plugin other than default, but once we have another batch plugin, we need to think about code reuse, and where the serialize logic for non batched request should live.

Reply:
i've already renamed it to transport-plugin, so this will be the place to hold all these logic. code sharing definitely needs to be considered if possible.

Tentatively, the interface would look like (written in Java flavor):
```java
interface TransportPlugin {
  string generateRequestUrl(baseUrl, segments);
 
  string generateBatchBodyPayload(baseUrl, segments);
  
  string generateSingleBodyPayload(baseUrl, segment);
}
```